test/unit/lib/three_scale/jobs_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ThreeScale::JobsTest < ActiveSupport::TestCase
+
+  def test_task_run
+    task = ThreeScale::Jobs::Task.new(String, :new, 'hello world')
+    task.run
+  end
```

Comment:
maybe add a test case that exceptions would be raised. e.g. `Integer, :sqrt, -4`

Reply:
The question would be: "raised by `what`?" 

-  `Integer.sqrt(-4)` raises an exception 
- or that `task.run` is aware of that exception?

There is no logic in my code that actively raises an exception. I do not think those kind of tests serve any purpose in understanding the logic, otherwise we can as well test any and all Exception like NoMethodError, RuntimeError etc .... and we will never be done testing.

I could probably test the `System::ErrorReporting` but it is just a capture the error, send it to a error service then re-raise it .... Dunno if it is good to have it here.

Reply:
The main benefit is to check behavior when a task raises some error, whatever it is.

My line of thinking is that tasks would raise errors from time to time. And we need to make sure that such occasions do not disturb our cron system as well they are properly reported in the log or elsewhere.

`Integer.sqrt(-4)` was just a suggested expression that would raise an error. Of course one can just use `raise "something"` :)

Reply:
But that is the goal of Sidekiq, ... if it raises an error then it will be re-run and you can see why it failed.
I really do not see the point checking that at the level of `Task` as it really does not have any error handling.

I would probably agree to have such test if you tell me we do not have error handling in the `Task` class and we need one. But in this case it is really on purpose. By design, I do not want to handle error: they are bubbled up to Sidekiq

Reply:
Cool. As you see fit.